python/tests/kernel/test_kernel_float_complex.py

```diff
@@ -0,0 +1,118 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+import os, sys
+
+import pytest
+import numpy as np
+import math
+from typing import List 
+
+import cudaq
+from cudaq import spin
+
+## [PYTHON_VERSION_FIX]
+skipIfPythonLessThan39 = pytest.mark.skipif(
+    sys.version_info < (3, 9),
+    reason="built-in collection types such as `list` not supported")
+
+def test_float():
+    """Test that we can use float numbers inside kernel functions."""
+
+    cudaq.reset_target()
+    cudaq.set_target('nvidia-fp64')
+
+    f = [0., 1., 1., 0.]
+
+    # Pass a list of float as a parameter
+    @cudaq.kernel
+    def test_float_vec_param(vec : list[float]):
+        f1 = vec
+
+    counts = cudaq.sample(test_float_vec_param, f)
```

Comment:
A better test here would be to define a `complex` return type on the kernel, and `assert test_float_vec_param(f) == that value`

something like 
```python
@cudaq.kernel
def test_float_vec_param(vec : list[float], i : int) -> complex: 
    f1 = vec 
    return vec[i]

for i in range(len(f)): 
   assert np.isclose(f[i], test_float_vec_param(f, i), atol=1e-6) 
```

Reply:
Changed the tests for now to work around this issue (will update the tests after the issue is fixed in the experimental branch as well)